### ISSUE TITLE
data labels are missing when you zoom

### ISSUE THREAD

**Issue by keithmdragon** [highcharts/highcharts#10930]
#### Expected behaviour
data labels are shown as expected when you zoom

#### Actual behaviour
data labels are missing when you zoom, even though there is plenty of room for them

#### Live demo with steps to reproduce
[live demo of issue](https://jsfiddle.net/kdragon/gd38rf7c/11/)

hit "ig", then hit "hg", u will notice that several points are not labeled that should be 

if u zoom to "hg" originally, it will show all the data labels, as it should

#### Product version
highcharts 7.1.2

#### Affected browser(s)
all

**Comment by KacperMadej** [comment]
Hi @keithmdragon 

Thank you for reporting about the problem.

Last working version 6.1.0

_Internal note:_ regression since 6.1.1

**Comment by sebastianbochan** [comment]
**Internal note:**
Bisect leads me to https://github.com/highcharts/highcharts/commit/f9361e5

**Comment by pawelfus** [comment]
**Workaround:**
Demo: http://jsfiddle.net/BlackLabel/ndys9wuL/

Snippet
```
(function(H) {
  H.addEvent(H.Series, 'afterDrawDataLabels', function() {
    this.points.forEach(function(point) {
      point.dataLabels.forEach(function(label) {
        label.hide = function() {
          label.attr({
            y: -9999
          });
          label.placed = false;
        };
      });
    });
  });
})(Highcharts)
```

*Internal note:*
Simplified demo: http://jsfiddle.net/BlackLabel/wq8n1ceL/

It happens when dataLabel was hidden by overlapping algorithm using `label.hide()`:
https://github.com/highcharts/highcharts/blob/7b23c336fd2e07269e75de1e466cda27a77188b7/js/modules/overlapping-datalabels.src.js#L223-L244
But `label.show()` won't be executed, because label is "hidden" in a safe way (label outside the extremes): https://github.com/highcharts/highcharts/blob/7b23c336fd2e07269e75de1e466cda27a77188b7/ts/parts/DataLabels.ts#L1350-L1354

**Comment by mounikay924** [comment]
Hi @TorsteinHonsi , what is the status of this issue? It causing the same issue with V6.1.1, data labels inside of bar chart are missing when zoom in or out.  should we need to upgrade to later version or need to add any code? Any help will be greatly appreciated

**Comment by KacperMadej** [comment]
Hi @mounikay924 
The bug is fixed since v7.1.3, so you need to update to at least that version.

Details: the bug was closed on 17 Jun 2019. In changelog - https://www.highcharts.com/blog/changelog/ - the first release after that date is v7.1.3 (2019-08-14). The fixed bug is also mentioned in the bug fixes for that version.
